src/Symfony/Component/DependencyInjection/Tests/Attribute/AutowireInlineTest.php

```diff
@@ -0,0 +1,200 @@
+        self::assertFalse($attribute->lazy);
+    }
+
+    /**
+     * @dataProvider provideFactories
+     */
+    public function testFactoryAndParamsLazy(string|array $factory, string|array $expectedResult)
+    {
+        $attribute = new AutowireInline($factory, ['someParam'], lazy: true);
+
+        $buildDefinition = $attribute->buildDefinition($attribute->value, null, $this->createReflectionParameterMock());
+
+        self::assertSame(null, $buildDefinition->getClass());
+        self::assertEquals($expectedResult, $buildDefinition->getFactory());
+        self::assertSame(['someParam'], $buildDefinition->getArguments());
+        self::assertTrue($attribute->lazy);
+    }
+
+    public static function provideFactories(): iterable
+    {
+        yield 'string callable' => [[null, 'someFunction'], [null, 'someFunction']];
+
+        yield 'class only' => [['someClass'], ['someClass', '__invoke']];
+        yield 'reference only' => [[new Reference('someClass')], [new Reference('someClass'), '__invoke']];
+
+        yield 'class with method' => [['someClass', 'someStaticMethod'], ['someClass', 'someStaticMethod']];
+        yield 'reference with method' => [[new Reference('someClass'), 'someMethod'], [new Reference('someClass'), 'someMethod']];
+    }
+
+    /**
+     * @dataProvider provideCalls
+     */
+    public function testCalls(string|array $calls, array $expectedResult)
+    {
+        $attribute = new AutowireInline('someClass', calls: $calls);
+
+        $buildDefinition = $attribute->buildDefinition($attribute->value, null, $this->createReflectionParameterMock());
+
+        self::assertSame('someClass', $buildDefinition->getClass());
+        self::assertSame($expectedResult, $buildDefinition->getMethodCalls());
+        self::assertSame([], $buildDefinition->getArguments());
+        self::assertFalse($attribute->lazy);
+    }
+
+    public static function provideCalls(): iterable
+    {
+        yield 'method with empty arguments' => [
+            [['someMethod', []]],
+            [['someMethod', []]],
+        ];
+        yield 'method with arguments' => [
+            [['someMethod', ['someArgument']]],
+            [['someMethod', ['someArgument']]],
+        ];
+        yield 'method without arguments with return clone true' => [
+            [['someMethod', [], true]],
+            [['someMethod', [], true]],
+        ];
+        yield 'method without arguments with return clone false' => [
+            [['someMethod', [], false]],
+            [['someMethod', []]],
+        ];
+        yield 'method with arguments with return clone true' => [
+            [['someMethod', ['someArgument'], true]],
+            [['someMethod', ['someArgument'], true]],
+        ];
+        yield 'method with arguments with return clone false' => [
+            [['someMethod', ['someArgument'], false]],
+            [['someMethod', ['someArgument']]],
+        ];
+    }
+
+    private function createReflectionParameterMock()
```

Comment:
we'd better not use a mock here - mocks are risky so we try to avoid them when possible

Reply:
[Done here](https://github.com/symfony/symfony/compare/4b0e7643ef8444649c452c490fc6335e390d324f..5955b18f44df3374fd1eb423e025f0a0f04d956c)